### PR TITLE
Add more comprehensive structs unit-tests

### PR DESCRIPTION
**Pull request by claudijd**
This change is [Reviewable](https://reviewable.io/reviews/certusone/wormhole/1006)

### CODE REVIEW

**Review by SEJeff — APPROVED** [review]
This is really good with 1 small caveat. You copy / paste `payload`, `governanceEmitter`, and `vaa` multiple times.

Perhaps make something vaguely like this?

```
func getVAA() *VAA {
	var payload = []byte{97, 97, 97, 97, 97, 97}
	var governanceEmitter = Address{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4}

	return &VAA{
		Version:          uint8(1),
		GuardianSetIndex: uint32(1),
		Signatures:       nil,
		Timestamp:        time.Unix(0, 0),
		Nonce:            uint32(1),
		Sequence:         uint64(1),
		ConsistencyLevel: uint8(32),
		EmitterChain:     ChainIDSolana,
		EmitterAddress:   governanceEmitter,
		Payload:          payload,
	}
```

That gives you a new copy each time and you don't need to worry about mutating the struct. Then your tests simply do:

```
vaa := getVAA()
...
```

That is also a light enough abstraction it doesn't impede readability.

File: node/pkg/vaa/structs_test.go [hunk 1]
```diff
@@ -123,3 +129,179 @@ func TestChainId_String(t *testing.T) {
...
+	h := vaa.SigningMsg()
+	pubKey, err := crypto.Ecrecover(h.Bytes(), vaa.Signatures[0].Signature[:])
+	assert.Nil(t, err)
+	assert.NotNil(t, pubKey)
+
+	// Translate that public key back to an address
+	addr := common.BytesToAddress(crypto.Keccak256(pubKey[1:])[12:])
+
+	// Make sure that it verifies
+	assert.True(t, vaa.VerifySignatures([]common.Address{addr}))
```
Comment (SEJeff) [inline node/pkg/vaa/structs_test.go:299]: Also this tests all of the positive cases, what what about the negative cases where there are errors? We probably want to cover those with a followup to this later on. Error code paths are always the least often used and most likely to break things.

Comment (claudijd) [reply node/pkg/vaa/structs_test.go:299]: I have another PR that goes into much greater depth on this topic.  This PR is largely just to establish a base from which to build an extensive suite of tests on top.

**Comment by claudijd** [comment]
@SEJeff great suggestions, used the abstraction technique and I will pile on to this pattern with a more extensive verifySignatures test suite.

**Review by bruce-riley — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/certusone/wormhole/1006#-:-MzL2yt0B2S6Ww9IrDD2:bnfp4nl)* status: all files reviewed, 1 unresolved discussion (waiting on @SEJeff)

**Review by SEJeff — APPROVED** [review]
